CPUMeter.c

```diff
@@ -55,6 +55,8 @@ static void CPUMeter_init(Meter* this) {
 
 // Custom uiName runtime logic to include the param (processor)
 static void CPUMeter_getUiName(const Meter* this, char* buffer, size_t length) {
+   assert(length > 0);
+
```

Comment:
Using `ATTR_NONNULL_ALL ATTR_ACCESS_RW(2,3)` would have the same effect at compile-time.

Similar with the other `*_UiName` functions.

Reply:
@BenBE Since `getUiName` functions are called through pointers, the function attribute would be best specified on pointers and not on individual functions.

And this could be the first time in htop codebase that we specify attributes on typedefs. I need a style to be decided, while in build tests all three styles work:

```c
// A
ATTR_NONNULL ATTR_ACCESS3_W(2, 3) typedef void(*Meter_GetUiName)(const Meter*, char*, size_t);
// B
typedef ATTR_NONNULL ATTR_ACCESS3_W(2, 3) void(*Meter_GetUiName)(const Meter*, char*, size_t);
// C
typedef void(*Meter_GetUiName)(const Meter*, char*, size_t) ATTR_NONNULL ATTR_ACCESS3_W(2, 3);
```

I would personally go with B, but I'd like to hear other people's opinions.

Reply:
I'd also suggest B, which is mot consistent with what we do for normal functions.